instrumentation/aws-sdk/aws-sdk-2.2/src/main/java8/io/opentelemetry/auto/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java

```diff
@@ -135,7 +132,7 @@ public void beforeTransmission(
     if (span != null) {
       // This scope will be closed by AwsHttpClientInstrumentation since ExecutionInterceptor API
       // doesn't provide a way to run code in the same thread after transmission has been scheduled.
-      ScopeHolder.CURRENT.set(currentContextWith(span));
+      ScopeHolder.CURRENT.set(ClientDecorator.currentContextWith(span));
```

Comment:
Do we still need this ScopeHolder? Why getting span from current context is not enough?

Reply:
We get the span from current context here, but `afterTransmission` may be called on a different thread, e.g., from netty, so the current context won't be accessible there. This `ScopeHolder` is only to allow closing in that method where we don't have context.

Reply:
And we don't have context propagation between here and Netty threads?

Reply:
Don't seem to yet. Here's the negative test we have in master

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/instrumentation/aws-sdk/aws-sdk-2.2/src/test/groovy/Aws2ClientTest.groovy#L223